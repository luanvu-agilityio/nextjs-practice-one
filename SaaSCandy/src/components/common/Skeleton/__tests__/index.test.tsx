import { render } from '@testing-library/react';
import { Skeleton } from '../index';

describe('Skeleton', () => {
  it('matches snapshot', () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { container } = render(<Skeleton className='custom-class' />);
    expect(container.firstChild).toHaveClass('custom-class');
  });

  it('has animate-pulse class', () => {
    const { container } = render(<Skeleton />);
    expect(container.firstChild).toHaveClass('animate-pulse');
  });
});
