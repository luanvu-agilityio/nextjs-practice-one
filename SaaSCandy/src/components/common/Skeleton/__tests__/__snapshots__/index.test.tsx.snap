// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Skeleton > matches snapshot 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-muted"
  />
</div>
`;

exports[`Skeleton matches snapshot 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-muted"
  />
</div>
`;
